test/install_test.go

```diff
@@ -206,13 +206,25 @@ func TestInject(t *testing.T) {
 
 func TestCheckProxy(t *testing.T) {
 	prefixedNs := TestHelper.GetTestNamespace("smoke-test")
+
+	// Tests Pods and Deployments
+	for deploy, replicas := range linkerdDeployReplicas {
+		if err := TestHelper.CheckPods(TestHelper.GetLinkerdNamespace(), deploy, replicas); err != nil {
+			t.Error(fmt.Errorf("Error validating pods for deploy [%s]:\n%s", deploy, err))
+		}
+		if err := TestHelper.CheckDeployment(TestHelper.GetLinkerdNamespace(), deploy, replicas); err != nil {
+			t.Error(fmt.Errorf("Error validating deploy [%s]:\n%s", deploy, err))
+		}
+	}
```

Comment:
This code unfortunately won't fix #1733. The failure described in that issue is caused by the linkerd-proxy containers in the "smoke-test" namespace not being in the ready state by the time `linkerd check --proxy` is run. But this code is only checking the pods in the "linkerd" namespace, not the "smoke-test" namespace.

Since you're adding the `--wait=0` flag, I think the easiest fix for the issue is to just retry running the check command until it succeeds. Something like this:

```
	var out string
	err := TestHelper.RetryFor(30*time.Second, func() error {
		var err error
		out, _, err = TestHelper.LinkerdRun(
			"check",
			"--proxy",
			"--expected-version",
			TestHelper.GetVersion(),
			"--namespace",
			prefixedNs,
			"--wait=0",
		)
		return err
	})
```

That's pretty similar to how `TestCheckPostInstall` is already setup in this file. Ideally this test and that test would be almost identical, save for the `--proxy` and `--namespace` flags. Speaking of which, we should add the `--wait=0` flag to the `TestCheckPostInstall` test, too.